x/evm/types/vfbc_denom_metadata.go

```diff
@@ -0,0 +1,59 @@
+package types
+
+import (
+	"fmt"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"math"
+)
+
+// VirtualFrontierBankContractDenomMetadata is a struct that contains the metadata of a contract denom,
+// collected from the bank module's denom metadata.
+type VirtualFrontierBankContractDenomMetadata struct {
+	MinDenom string `json:"min_denom"` // corresponds to the "base" in the bank denom metadata
+	Decimals uint32 `json:"decimals"`  // decimals is the biggest "exponent" among units of bank denom metadata
+	Name     string `json:"name"`      // corresponds to the "display" (fallback to "name") in the bank denom metadata
+	Symbol   string `json:"symbol"`    // corresponds to the "symbol" in the bank denom metadata
+}
+
+// CollectMetadataForVirtualFrontierBankContract collects the metadata of a contract denom from the bank module's denom metadata,
+// to be used to return result for the Virtual Frontier Bank Contract execution.
+func CollectMetadataForVirtualFrontierBankContract(
+	metadata banktypes.Metadata,
+) (
+	vfbcDenomMeta VirtualFrontierBankContractDenomMetadata,
+	isInputPassValidation bool,
+) {
+	isInputPassValidation = metadata.Validate() == nil
```

Comment:
why not return here on `!isInputPassValidation`?

Reply:
@mtsitrin this is decision made by @omritoptix 

> well  I mean that validate upon creation of the contract, but when returning, if the data is now invalid cause it was updated, simply return what’s there.
> and than worst case we fix it

The flag of invalid will be ignored during execution.